src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,573 @@
+        """
+        st = self.simplex_tree
+        f = open(self.input_name + ".txt", "w")
+        f.write(self.input_name + "\n")
+        f.write(self.cover_name + "\n")
+        f.write(self.color_name + "\n")
+        f.write("0 0\n")
+        f.write(str(st.num_vertices()) + " " + str(len(list(st.get_skeleton(1)))-st.num_vertices()) + "\n")
+        name2id = {}
+        idv = 0
+        for s,_ in st.get_skeleton(0):
+            f.write(str(idv) + " " + str(self.node_info[s[0]]["colors"][0]) + " " + str(self.node_info[s[0]]["size"]) + "\n")
+            name2id[s[0]] = idv
+            idv += 1
+        for s,_ in st.get_skeleton(1):
+            if len(s) == 2:
+                f.write(str(name2id[s[0]]) + " " + str(name2id[s[1]]) + "\n")
+        f.close()
+
+    class _constant_clustering():
+        def fit_predict(X):
+            return np.zeros([len(X)], dtype=np.int32)
+
+
+class MapperComplex(CoverComplexPy, BaseEstimator, TransformerMixin):
+    """
+    This is a class for computing Mapper simplicial complexes on point clouds or distance matrices.
+    """
+    def __init__(self, input_type="point cloud", colors=None, mask=0,
+                       filters=None, filter_bnds=None, resolutions=None, gains=None, clustering=DBSCAN(), N=100, beta=0., C=10.,
+                       input_name="data", cover_name="cover", color_name="color", verbose=False):
+        """
+        Constructor for the MapperComplex class.
+
+        Parameters:
+            input_type (string): type of input data. Either "point cloud" or "distance matrix".
+            colors (numpy array of shape (num_points) x (num_colors)): functions used to color the nodes of the cover complex. More specifically, coloring is done by computing the means of these functions on the subpopulations corresponding to each node. If None, first coordinate is used if input is point cloud, and eccentricity is used if input is distance matrix.
+            mask (int): threshold on the size of the cover complex nodes (default 0). Any node associated to a subpopulation with less than **mask** points will be removed.
+            filters (numpy array of shape (num_points) x (num_filters)): filter functions (sometimes called lenses) used to compute the cover. Each column of the numpy array defines a scalar function defined on the input points.
+            filter_bnds (numpy array of shape (num_filters) x 2): limits of each filter, of the form [[f_1^min, f_1^max], ..., [f_n^min, f_n^max]]. If one of the values is numpy.nan, it can be computed from the dataset with the fit() method.
+            resolutions (numpy array of shape num_filters containing integers): resolution of each filter function, ie number of intervals required to cover each filter image. If None, it is estimated from data.
+            gains (numpy array of shape num_filters containing doubles in [0,1]): gain of each filter function, ie overlap percentage of the intervals covering each filter image.
```

Comment:
I expect it is common to use `np.broadcast_to(frac, n)` for gains. Would it make sense to allow passing a `float` and automatically expanding it to the relevant array in that case?

Reply:
Yep, I've added this possibility.